### DISCUSSION TITLE
Prisma migration in Docker: Production

### DISCUSSION THREAD

**Discussion by ahmed-trailfive** [prisma/prisma#13809] [Q&A]
Hi! Need some information about running migrations in Docker.
Database: Google cloud MySQL

I am using the following script.

```
FROM node:16-alpine
WORKDIR /app
COPY package.json ./
COPY tsconfig.json ./
COPY prisma ./prisma/
EXPOSE 4000
RUN npm install -g ts-node && npm install 
COPY . .

RUN npx prisma migrate deploy 

CMD ["npm", "run", "dev"]
```

Prisma is not able to connect with the MySQL as only the "instance `A` that runs the docker image" has the permissions.

Where should I run the migrations?

**Comment by ruheni** [comment]
Hey 👋🏽

Suppose your database is running on a different service, you will need to include/expose the `DATABASE_URL` connection string in your Dockerfile – which you would configure in the deployment platform (e.g., Cloud Run) as part of the environment variables.

Here's an example of the modification you would make to your script:

```diff
FROM node:16-alpine
WORKDIR /app
COPY package.json ./
COPY tsconfig.json ./
COPY prisma ./prisma/
EXPOSE 4000
RUN npm install -g ts-node && npm install 
COPY . .

+ENV DATABASE_URL=$DATABASE_URL

RUN npx prisma migrate deploy 

CMD ["npm", "run", "dev"]
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
